core/src/main/scala/org/apache/spark/api/r/RRunner.scala

```diff
@@ -40,7 +40,8 @@ private[spark] class RRunner[U](
     broadcastVars: Array[Broadcast[Object]],
     numPartitions: Int = -1,
     isDataFrame: Boolean = false,
-    colNames: Array[String] = null)
+    colNames: Array[String] = null,
+    key: Array[String] = null)
```

Comment:
remove this useless parameter. No need to pass grouping column names to R worker